autoblocks/_impl/testing/run.py

```diff
@@ -52,6 +52,14 @@ def serialize(x: Any) -> Any:
     return orjson.loads(orjson.dumps(x, default=orjson_default))
 
 
+async def all_settled(coroutines: list[Coroutine[Any, Any, Any]]) -> None:
+    """
+    Runs all the coroutines in parallel and waits for all of them to finish,
+    regardless of whether they succeed or fail. Similar to Promise.allSettled.
+    """
+    await asyncio.gather(*coroutines, return_exceptions=True)
+
```

Comment:
Could we potentially reuse `gather_with_max_concurrency` from `util.py` or consolidate?

Reply:
this doesn't have a max concurrency and i don't think we should modify `gather_with_max_concurrency` to handle situations w/ no max concurrency, since that is all it does

Reply:
i'm also going to delete it anyway when both your PR and https://github.com/autoblocksai/python-sdk/pull/128 are merged b/c we'll no longer have the `max_evaluator_concurrency` arg